### PR TITLE
add version, domain and workstation fields

### PR DESCRIPTION
**Pull request by justdan96**
Add version support from jh125486
Add workstation and domain support from jh125486
Enable to extract domain from username

Contributing pull request. Most of the heavy lifting was done by jh125486, I just added a few lines to read the domain from the username. This is required so our team can use domain credentials for logon in masterzen/winrm.

### CODE REVIEW

File: negotiator.go [hunk 1]
```diff
@@ -76,8 +77,20 @@ func (l Negotiator) RoundTrip(req *http.Request) (res *http.Response, err error)
 			return nil, err
 		}
 
+		// parse domain name from username
+		domain := ""
+
+		if strings.Contains(u, "\\") {
+			ucomponents := strings.Split(u, "\\")
```
Comment (paulmey) [inline negotiator.go]: use [SplitN](https://golang.org/pkg/strings/#SplitN)?

File: negotiate_message.go [hunk 2]
```diff
@@ -3,29 +3,61 @@ package ntlmssp
...
+		Workstation:    newVarField(&payloadOffset, len(workstationName)),
+		Version:        DefaultVersion(),
+	}
 
 	b := bytes.Buffer{}
-	err := binary.Write(&b, binary.LittleEndian, &m)
-	if err != nil {
-		panic(err)
+	if err := binary.Write(&b, binary.LittleEndian, &msg); err != nil {
+		return nil, err
```
Comment (paulmey) [inline negotiate_message.go]: Thanks for fixing my panic... ;-)

**Comment by paulmey** [comment]
can we have some tests for the new functionality?

**Comment by justdan96** [comment]
I wasn't sure exactly what to add for the testing so hopefully what I've done is okay. SplitN is in there as well, I ended up creating a new func for parsing the username to make things simpler. I might look at checking if the username is valid and putting it into the same function, if I can find a decent spec for what is considered a valid username.

**Comment by paulmey** [comment]
This way, we at least have some tests fail if the structure of the message changes.

**Comment by justdan96** [comment]
Thanks!

**Pull request merged**